ddln_imaginator.proto

```diff
@@ -21,6 +21,64 @@ enum ImageFormat {
     GRAY = 4;
 }
 
+message Environment {
+    message Clouds {
+        enum Type {
+            NONE = 0;
+            ALTOCUMULUS = 1;
+            CLOUDS_BASE = 2;
+            CUMULUS = 3;
+            NIMBOSTRATUS = 4;
+            STRATOCUMULUS = 5;
+            STRATUS = 6;
+        }
+        optional Type type = 1;
+        optional float altitude_m = 2;
+        optional Point3d velocity = 3; // Only use the directional x,y,z values to set the wind in the Engine.
+    }
+    optional Clouds clouds = 1;
+
+    message Haze {
+        enum Type {
+            DEFAULT = 0; // Visibility = 20000, density = 1
+            CUSTOM = 1; // Custom visibility and density as specified below
+        }
+        optional Type type = 1;
+        optional float visibility = 2; // recommended 0 - 20000
+        optional float density = 3; // recommended 0.001 - 100
+    }
+    optional Haze haze = 2;
+
+    message Fog {
+        optional bool enabled = 1;
+        optional float multiplier = 2 ; // recommended 0.01 - 0.0001
+    }
+    optional Fog fog = 3;
+
+    message Sun {
+        enum Type {
+            DEFAULT = 0;
```

Comment:
Add comments to specify default and custom behavior.

Reply:
DEFAULT = 0; // Keeps the sun attached with the world. Parameters are not changeable.
CUSTOM = 1; // Customized sun with parameters below